test/e2e/provisioner/provisioner_test.go

```diff
@@ -91,62 +93,139 @@ var _ = Describe("Gateway provisioner", func() {
 			}
 
-			_, ok := f.CreateGatewayAndWaitFor(gateway, gatewayReady)
+			gateway, ok := f.CreateGatewayAndWaitFor(gateway, func(gw *gatewayapi_v1alpha2.Gateway) bool {
+				return gatewayReady(gw) && gatewayHasAddress(gw)
+			})
 			require.True(f.T(), ok)
-		})
-	})
 
-	f.NamespacedTest("multiple-gateways-per-namespace", func(namespace string) {
-		Specify("Multiple basic one-listener HTTP gateways can be provisioned in a single namespace", func() {
-			gateway := &gatewayapi_v1alpha2.Gateway{
+			f.Fixtures.Echo.Deploy(namespace, "echo")
+
+			route := &gatewayapi_v1alpha2.HTTPRoute{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "http-1",
 					Namespace: namespace,
+					Name:      "httproute-1",
 				},
-				Spec: gatewayapi_v1alpha2.GatewaySpec{
-					GatewayClassName: gatewayapi_v1alpha2.ObjectName("contour"),
-					Listeners: []gatewayapi_v1alpha2.Listener{
+				Spec: gatewayapi_v1alpha2.HTTPRouteSpec{
+					Hostnames: []gatewayapi_v1alpha2.Hostname{"provisioner.projectcontour.io"},
+					CommonRouteSpec: gatewayapi_v1alpha2.CommonRouteSpec{
+						ParentRefs: []gatewayapi_v1alpha2.ParentRef{
+							gatewayapi.GatewayParentRef("", gateway.Name),
+						},
+					},
+					Rules: []gatewayapi_v1alpha2.HTTPRouteRule{
 						{
-							Name:     "http",
-							Protocol: gatewayapi_v1alpha2.HTTPProtocolType,
-							Port:     gatewayapi_v1alpha2.PortNumber(80),
-							AllowedRoutes: &gatewayapi_v1alpha2.AllowedRoutes{
-								Namespaces: &gatewayapi_v1alpha2.RouteNamespaces{
-									From: gatewayapi.FromNamespacesPtr(gatewayapi_v1alpha2.NamespacesFromSame),
-								},
-							},
+							Matches:     gatewayapi.HTTPRouteMatch(gatewayapi_v1alpha2.PathMatchPathPrefix, "/prefix"),
+							BackendRefs: gatewayapi.HTTPBackendRef("echo", 80, 1),
 						},
 					},
 				},
 			}
-
-			_, ok := f.CreateGatewayAndWaitFor(gateway, gatewayReady)
+			_, ok = f.CreateHTTPRouteAndWaitFor(route, httpRouteAccepted)
 			require.True(f.T(), ok)
 
-			gateway = &gatewayapi_v1alpha2.Gateway{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "http-2",
-					Namespace: namespace,
-				},
-				Spec: gatewayapi_v1alpha2.GatewaySpec{
-					GatewayClassName: gatewayapi_v1alpha2.ObjectName("contour"),
-					Listeners: []gatewayapi_v1alpha2.Listener{
-						{
-							Name:     "http",
-							Protocol: gatewayapi_v1alpha2.HTTPProtocolType,
-							Port:     gatewayapi_v1alpha2.PortNumber(80),
-							AllowedRoutes: &gatewayapi_v1alpha2.AllowedRoutes{
-								Namespaces: &gatewayapi_v1alpha2.RouteNamespaces{
-									From: gatewayapi.FromNamespacesPtr(gatewayapi_v1alpha2.NamespacesFromSame),
+			res, ok := f.HTTP.RequestUntil(&e2e.HTTPRequestOpts{
+				OverrideURL: "http://" + gateway.Status.Addresses[0].Value,
+				Host:        string(route.Spec.Hostnames[0]),
+				Path:        "/prefix/match",
+				Condition:   e2e.HasStatusCode(200),
+			})
+			require.Truef(f.T(), ok, "expected 200 response code, got %d", res.StatusCode)
+			require.NotNil(f.T(), res)
```

Comment:
nit: might need to reverse the order of 132 and 133 since the error message uses a field in the `res` variable

Reply:
ah yeah good call!